templates/typescript-testing/test/restate_test_environment.ts

```diff
@@ -109,6 +110,68 @@ export class RestateTestEnvironment {
         )}`;
     }
 
+    public async setState(
+      service: restate.VirtualObjectDefinition<any, any> | restate.WorkflowDefinition<any, any>,
+      key: string, 
+      newState: {[key: string]: any}) {
+      const res = await fetch(
+          `${this.adminAPIBaseUrl()}/services/${service.name}/state`,
+          {
+              method: "POST",
+              headers: {
+                  "Content-Type": "application/json",
+              },
+              body: JSON.stringify({
+                  object_key: key,
+                  // the endpoint expects a map of key -> bytes as JSON array of numbers
+                  new_state: Object.fromEntries(Object.entries(newState).map(([key, value]) => {
+                    const valueJSON = new TextEncoder().encode(JSON.stringify(value))
+                    
+                    return [key, Array.from(valueJSON)]
+                  })),
+              }),
+          }
+      );
+
+      if (!res.ok) {
+          const badResponse = await res.text();
+          throw new Error(
+              `Error ${res.status} during modify state: ${badResponse}`
+          );
+      }
+    }
```

Comment:
for reference, In Java I just generate a client from our OpenAPI definition and publish it as separate module, and then use it in our `sdk-testing` module (the one with the testcontainers thing)

Reply:
What I would hope is that in typescript SDK we eventually have an autogenerated package for the admin api. But yes you can ignore this comment now, nothing we should be doing for it.